File: AngularHotelManagementCodegen/src/app/services/HotelContractDTO .ts
import { Hotel } from "./hotel";
import { SeasonDTO } from './SeasonDTO';
import { Supplement } from "./supplement";
import { SupplementPriceDTO } from "./SupplementPriceDTO";
import { MarkupDTO } from "./MarkupDTO";
import { DiscountDTO } from "./DiscountDTO";
import { RoomTypeDTO } from "./RoomTypeDTO";
import { RoomTypePriceDTO } from "./RoomTypePriceDTO";

export interface HotelContractDTO {
    contractID: number;
    hotelID: number;
    startDate: Date;
    endDate: Date;
    termsAndConditions: string;
    hotel: Hotel; 
    seasons: SeasonDTO[];
    supplements: Supplement[];
    supplementPrices: SupplementPriceDTO[];
    markupPrices: MarkupDTO[];
    discounts: DiscountDTO;
    roomTypes: RoomTypeDTO[];
    roomTypePrices: RoomTypePriceDTO[];
  }

File: HotelSpringBootCodegen/hotel/src/main/java/com/hotel/hotel/DTO/HotelContractDTO.java
package com.hotel.hotel.DTO;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.hotel.hotel.entity.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class HotelContractDTO {

    private int contractID;
    @JsonSerialize(using = CustomDateSerializer.class)
    private Date startDate;
    @JsonSerialize(using = CustomDateSerializer.class)
    private Date endDate;
    private String termsAndConditions;

    private HotelDTO hotel;
    private Set<SeasonDTO> seasons;
    private Set<SupplementDTO> supplements;
    private Set<SupplementPriceDTO> supplementPrices;

    private Set<MarkupDTO> markupPrices;

    private DiscountDTO discounts;

    private Set<RoomTypeDTO> roomTypes;

    private Set<RoomTypePriceDTO> roomTypePrices;
}
